backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/DocLine_Allocation.java

```diff
@@ -643,4 +632,30 @@ public boolean isPaymentReceipt()
 		return paymentReceipt;
 	}
 
+	public boolean isValidPurchaseSalesCompensationAmt()
+	{
+		final BigDecimal compensationAmtSource = getAllocatedAmt();
+		final DocLine_Allocation counterLine = getCounterDocLine();
+		final BigDecimal counterLine_compensationAmtSource = counterLine.getAllocatedAmt();
+
+		if ((isAPI() && counterLine.isARC()) || (isARC() && counterLine.isAPI()))
+		{
+			return compensationAmtSource.compareTo(counterLine_compensationAmtSource) == 0;
+		}
+		else
+		{
+			return compensationAmtSource.compareTo(counterLine_compensationAmtSource.negate()) == 0;
+		}
+	}
+
+	private boolean isARC()
+	{
+		return isCreditMemoInvoice() && !isSOTrxInvoice();
```

Comment:
imho here it needs to be isSOTrxInvoice()

also, for my ODD: pls keep the same order of checking in both isAPI and is ARC,
i.e. check isSOTrx first  && check Credit Memo